src/core/utility/file/filePath/FilePath.cpp

```diff
@@ -2,59 +2,49 @@
 
 #include <regex>
 
-#include <boost/filesystem.hpp>
-#include <boost/filesystem/path.hpp>
-
+#include "FileSystem.h"
 #include "logging.h"
 #include "utilityString.h"
 
-FilePath::FilePath()
-    : m_path(std::make_unique<boost::filesystem::path>(""))
-    , m_exists(false)
-    , m_checkedExists(false)
-    , m_isDirectory(false)
-    , m_checkedIsDirectory(false)
-    , m_canonicalized(false) {}
-
-FilePath::FilePath(const std::string& filePath)
-    : m_path(std::make_unique<boost::filesystem::path>(filePath))
-    , m_exists(false)
-    , m_checkedExists(false)
-    , m_isDirectory(false)
-    , m_checkedIsDirectory(false)
-    , m_canonicalized(false) {}
-
-FilePath::FilePath(const std::wstring& filePath)
-    : m_path(std::make_unique<boost::filesystem::path>(filePath))
-    , m_exists(false)
-    , m_checkedExists(false)
-    , m_isDirectory(false)
-    , m_checkedIsDirectory(false)
-    , m_canonicalized(false) {}
+FilePath::FilePath() : m_path(std::make_unique<fs::path>()), m_canonicalized(false) {}
+
+FilePath::FilePath(const std::string& filePath) : m_path(std::make_unique<fs::path>(filePath)), m_canonicalized(false) {}
+
+FilePath::FilePath(const std::wstring& filePath) : m_path(std::make_unique<fs::path>(filePath)), m_canonicalized(false) {}
```

Comment:
Please define m_canonicalized = false in the header

Reply:
I agree.